code/game/machinery/telecomms/telecomunications.dm

```diff
@@ -171,3 +172,49 @@ GLOBAL_LIST_EMPTY(telecomms_list)
 
 /obj/machinery/telecomms/proc/de_emp()
 	stat &= ~EMPED
+
+//================
+// TELECOMMUNICATIONS MAP SAVE
+// Telecomm networks on a saved map will autolink when spawned in!
+//================
+
+/obj/machinery/telecomms/get_pre_save_key()
+	return network
+
+//Run through all machines in the group and generate autolinkers
+/obj/machinery/telecomms/pre_save(list/group, pre_save_key = "")
+	//Makes it very rare that 2 maps will have the same autolinkers
+	var/map_key = rand(1, 999999)
```

Comment:
> not truly unique

😩

Reply:
well I could use [GLOB.round_id]_[saves] or something. The chance of 2 having the same key is like 1/100000 and if they do then 2 machines will link but not send data due to being on different z-levels. Wouldn't be too hard to do though in fairness.